docs/releasing/testing-and-linting.md

```diff
@@ -2,55 +2,88 @@
+
+    `npm test -- --watch src/govuk/components/button`
+
+### Running only SASS linting
+
+```
+npm run lint:scss
+```
 
 See [CSS Coding Standards](/docs/contributing/coding-standards/css.md#linting) for details.
 
-## Javascript linting
+### Running only JavaScript linting
+
+```
+npm run lint:js
+```
 
 See [JavaScript Coding Standards](/docs/contributing/coding-standards/js.md#formatting-and-linting) for details.
 
 ## Unit and functional tests with Node.js
 
 We use [Jest](https://jestjs.io/), an automated testing platform with an assertion library, and [Puppeteer](https://pptr.dev/) that is used to control [headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome).
 
-We test individual components and for instance global Sass files. See [all.test.js](../../src/govuk/all.test.js). for examples of global tests we run, and  [checkboxes.test.js](../../src/govuk/components/checkboxes/checkboxes.test.js) for an example of component tests.
+### Component tests
+We write functional tests for every component to check the output of our Nunjucks code. These are found in `template.test.js` files in each component directory. These Nunjucks tests call `render` on the component examples defined in the component yaml files, and asserts the HTML tags, attributes and classes are as expected. For example: checking that when you pass in an `id` to the component using the Nunjucks macro, it outputs the component with an `id` attribute equal to that value.
 
-We aim to write the description of our tests in as "natural language" as possible, for instance "back-link component fails to render if the required fields are not included".
+If a component uses JavaScript, we also write functional tests in a `[component name].test.js` file, for example [checkboxes.test.js](../../src/govuk/components/checkboxes/checkboxes.test.js). These component tests check that interactions, e.g: a mouse click, have the expected result, e.g: the component gaining focus or the addition of a class/HTML attribute.
 
-### Running individual tests
-You can run a subset of the test suite that only tests components by running:
+Component JavaScript logic should also be tested using unit tests, in a `[component name].unit.test.mjs` file. These tests are better suited for testing behind-the-scenes logic or in cases where the final output of some logic isn't a change to the component markup.
 
-    `npm test -- src/govuk/components/button`
+### Global tests
+We write functional tests for checking our JavaScript exports and our global sass variables - see [all.test.js](../../src/govuk/all.test.js) for examples of global tests we run.
 
-Note: There's a watch mode that keeps a testing session open waiting for changes that can be used with:
+### Conventions
+We aim to write the description of our tests in as "natural language" as possible, for instance "back-link component fails to render if the required fields are not included".
```

Comment:
Would changing 'natural language' still work if it was changed to 'plain English'? Something like this - We aim to write the test descriptions in plain English. For example, "back-link...etc

Reply:
I'm not sure - do you think that has the same meaning? I think we're trying to say that test names should make sense as a sentence when read out loud, which may be slightly different to plain English. Although I'm struggling to think of an example 🤔

Reply:
Ah I see. Well 'colloquial' would be an accurate word, but not particularly user friendly. How about 'everyday language' or 'conversational language'?

Reply:
@claireashworth Silly question - how would you word that as a sentence? E.g: "we aim to write the test descriptions in everyday language" - would that be ok?

Reply:
yup that sounds good